### PR TITLE
Cr129 support survey launched endpoint and send message to case service

### PR DESCRIPTION
**Pull request by PeterBochel**
# Motivation and Context
This change supports the receipt of an http request which states that a survey has been launched. This results in a SurveyLaunched message being written to a RabbitMQ exchange.

# What has changed
- This change has resulted in the creation of a RestController class (RespondentHomeEndpoint.java) to implement the "/SurveyLaunched" endpoint.
- DTO classes have also been created, to hold both the initial request and the content of the message which is written to the RabbitMQ exchange.
- Unit and integration tests have been added.

# How to test?
To test automatically run the unit and integration tests.

# How to manually test?
1 - In the RH project Edit src/main/resources/springintegration/broker.xml
Find the "\<!-- Start of Queues --\>"
Add the following to define the queues:
>>>
	<rabbit:queue name="Case.SurveyLaunched" durable="true">
		<rabbit:queue-arguments value-type="java.lang.String">
			<entry key="x-dead-letter-exchange" value="case-deadletter-exchange" />
			<entry key="x-dead-letter-routing-key" value="Case.Gateway.binding" />
		</rabbit:queue-arguments>
	</rabbit:queue>

 	<rabbit:queue name="Case.SurveyLaunchedDLQ" durable="true" />

2 - Then in the same file. Find the '\<!-- Start of Exchanges --\>' section.
Add to the bindings for the 'events' exchange.
`
	<rabbit:binding queue="Case.SurveyLaunched" key="event.response.authentication" />
`

3 - Create file with request body:
`
cat > /tmp/surveyLaunchedDTO.json << EOF
{
  "questionnaireId": "23434234234",
  "caseId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
}
EOF
`

4 - Start the RH server and submit a survey Launched post request:
``
$ curl -s -X POST -H "Content-Type: application/json" --data @/tmp/surveyLaunchedDTO.json --user admin:secret "http://localhost:8171/surveyLaunched"
``

5 - Check the content of the Case.surveyLaunched queue.

# Links
See CR: https://collaborate2.ons.gov.uk/jira/browse/CR-129

### CODE REVIEW

File: src/main/resources/springintegration/flows.xml [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <int:annotation-config/>
 
     <import resource="case-gateway-event-inbound.xml"/>
```
Comment (philwhiles) [inline src/main/resources/springintegration/flows.xml]: consistent naming here please

File: src/main/resources/springintegration/flows.xml [hunk 2]
```diff
@@ -9,6 +9,7 @@
...
+    <import resource="case-service-event-outbound.xml"/>
     <import resource="respondent-service-event-outbound.xml"/>
```
Comment (philwhiles) [inline src/main/resources/springintegration/flows.xml]: remove - but look for vestigial refs

File: src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentHomeServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    surveyLaunchedDTO.setCaseId(UUID.fromString("3fa85f64-5717-4562-b3fc-2c963f66afa6"));
+    respondentHomeService.surveyLaunched(surveyLaunchedDTO);
+
+    // Get hold of the event that respondentHomeService created
+    Mockito.verify(publisher).sendEvent(sendEventCaptor.capture());
+    SurveyLaunchedEvent surveyLaunchedEvent = sendEventCaptor.getValue();
+
+    // Verify the contents of the top level event data
+    assertEquals("SURVEY_LAUNCHED", surveyLaunchedEvent.getEvent().getType());
+    assertEquals("CONTACT_CENTRE_API", surveyLaunchedEvent.getEvent().getSource());
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentHomeServiceImplTest.java]: Ahem!!!

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentHomeServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            + surveyLaunchedDTO.getQuestionnaireId()
+            + ", caseId: "
+            + surveyLaunchedDTO.getCaseId()
+            + ". Using transactionId: "
+            + transactionId);
+
+    // Build key parts of Survey Launched event
+    Header eventData =
+        Header.builder()
+            .type("SURVEY_LAUNCHED")
```
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentHomeServiceImpl.java]: Updated code to use RH constants

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentHomeServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        "Generating SurveyLaunched event for questionnaireId: "
+            + surveyLaunchedDTO.getQuestionnaireId()
+            + ", caseId: "
+            + surveyLaunchedDTO.getCaseId()
+            + ". Using transactionId: "
+            + transactionId);
+
+    // Build key parts of Survey Launched event
+    Header eventData =
+        Header.builder()
```
[the remark this answers is not in the document]
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentHomeServiceImpl.java]: Moved hardcoded string to the existing Constants class

File: src/main/resources/springintegration/flows.xml (hunk 2 above)
Comment (PeterBochel) [reply src/main/resources/springintegration/flows.xml]: Removed config xml file. 
Also removed some unused code which was using the rabbit template bean.

File: src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentHomeServiceImplTest.java (hunk 3 above)
Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/RespondentHomeServiceImplTest.java]: Removed

File: src/main/resources/springintegration/flows.xml (hunk 1 above)
Comment (PeterBochel) [reply src/main/resources/springintegration/flows.xml]: Names of 3 included files renamed. 
Please review.

**Pull request merged**